OsmAndMapCreator/src/net/osmand/BinaryMerger.java

```diff
@@ -286,6 +302,71 @@ private void combineAddressIndex(String name, BinaryMapIndexWriter writer, Addre
 		writer.endWriteAddressIndex();
 	}
 
+	private void combinePoiIndex(String name, BinaryMapIndexWriter writer, long dateCreated, PoiRegion[] poiRegions, BinaryMapIndexReader[] indexes)
+			throws IOException, SQLException {
+		int readPoiCount = 0;
+		int badNeg = 0;
+//		int writtenPoiCount = 0;
+		MapRenderingTypesEncoder renderingTypes = new MapRenderingTypesEncoder(null, name);
+		boolean overwriteIds = true;
+		IndexPoiCreator indexPoiCreator = new IndexPoiCreator(renderingTypes, overwriteIds);
+		indexPoiCreator.createDatabaseStructure(new File(new File(System.getProperty("user.dir")), IndexCreator.getPoiFileName(name)));
+		List<Long> ids = new ArrayList<Long>();
+		Amenity a = null;
+		for (int i = 0; i < poiRegions.length; i++) {
+			BinaryMapIndexReader index = indexes[i];
+			log.info("Region: " + extractRegionName(index));
+			List<Amenity> amenities = index.searchPoi(BinaryMapIndexReader.buildSearchPoiRequest(
```

Comment:
This is a wrong way to query all amenities because they are all load in to memory, instead ResultMatcher should be used for streaming approach and returning "false" to not keep object in memory.